api/v1alpha1/search_types.go

```diff
@@ -94,69 +100,33 @@ type DeploymentConfig struct {
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 
 	// +optional
-	//Image_override
+	// Image_override
 	ImageOverride string `json:"imageOverride,omitempty"`
 
 	// +optional
-	//ImagePullSecret
-	ImagePullSecret string `json:"imagePullSecret,omitempty"`
-
-	//ImagePullPolicy
-	// +optional
-	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
-
-	// NodeSelector to schedule on nodes with matching labels
-	// +optional
-	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
-
-	//Proxy config , if remote collectors need to override
-	// +optional
-	ProxyConfig map[string]string `json:"proxyConfig,omitempty"`
+	// Container Arguments
+	Arguments []string `json:"arguments,omitempty"`
 }
 
 type StorageSpec struct {
-	// +optional
 	// name of the storage class
 	StorageClassName string `json:"storageClassName,omitempty"`
 	// +optional
-	// storage capacity
+	// +kubebuilder:default:="10Gi"
```

Comment:
I don't know how to do it from here, but in V1 we define a regular expression to validate the values. Should we add that here?

Reply:
Size type is  *resource.Quantity - That will auto check , you cant include invalid values

Reply:
Good, auto check is much better! :-)